tests/subsys/logging/log_core_additional/src/log_test.c

```diff
@@ -0,0 +1,396 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Additional test case for log core
+ *
+ */
+
+#include <tc_util.h>
+#include <stdbool.h>
+#include <zephyr.h>
+#include <ztest.h>
+#include <logging/log_backend.h>
+#include <logging/log_ctrl.h>
+#include <logging/log.h>
+
+#define LOG_MODULE_NAME log_test
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, LOG_LEVEL_INF);
+
+struct backend_cb {
+	size_t counter;
+	size_t sync;
+	bool check_timestamp;
+	uint32_t exp_timestamps[16];
+	bool check_severity;
+	uint16_t exp_severity[4];
+	bool check_domain_id;
+};
+
+static void put(struct log_backend const *const backend, struct log_msg *msg)
+{
+	log_msg_get(msg);
+	struct backend_cb *cb = (struct backend_cb *)backend->cb->ctx;
+
+	if (cb->check_domain_id) {
+		zassert_equal(log_msg_domain_id_get(msg), CONFIG_LOG_DOMAIN_ID,
+				"Unexpected domain id");
+	}
+
+	if (cb->check_timestamp) {
+		uint32_t exp_timestamp = cb->exp_timestamps[cb->counter];
+
+		zassert_equal(log_msg_timestamp_get(msg), exp_timestamp,
+			      "Unexpected message index");
+	}
+
+	if (cb->check_severity) {
+		zassert_equal(log_msg_level_get(msg),
+			      cb->exp_severity[cb->counter],
+			      "Unexpected log severity");
+	}
+
+	cb->counter++;
+	log_msg_put(msg);
+}
+
+static void sync_string(const struct log_backend *const backend,
+		     struct log_msg_ids src_level, uint32_t timestamp,
+		     const char *fmt, va_list ap)
+{
+	struct backend_cb *cb = (struct backend_cb *)backend->cb->ctx;
+
+	cb->counter++;
+	cb->sync++;
+}
+
+static void sync_hexdump(const struct log_backend *const backend,
+			 struct log_msg_ids src_level, uint32_t timestamp,
+			 const char *metadata, const uint8_t *data,
+			 uint32_t length)
+{
+	struct backend_cb *cb = (struct backend_cb *)backend->cb->ctx;
+
+	cb->counter++;
+	cb->sync++;
+}
+
+const struct log_backend_api log_backend_test_api = {
+	.put = put,
+	.put_sync_string = IS_ENABLED(CONFIG_LOG_IMMEDIATE) ?
+			sync_string : NULL,
+	.put_sync_hexdump = IS_ENABLED(CONFIG_LOG_IMMEDIATE) ?
+			sync_hexdump : NULL,
+	.panic = NULL,
+	.dropped = NULL,
+	.init = NULL,
+};
+
+LOG_BACKEND_DEFINE(backend1, log_backend_test_api, false);
+struct backend_cb backend1_cb;
+
+LOG_BACKEND_DEFINE(backend2, log_backend_test_api, false);
+struct backend_cb backend2_cb;
+
+static uint32_t stamp;
```

Comment:
/* Counts number of times the timestamp is read */

or some sort of comment

Reply:
Updated